app/services/budgetkey-item.ts

```diff
@@ -46,22 +46,42 @@ export class BudgetKeyItemService {
     Promise.reject(new Error('No layout for ' + path));
   }
 
-  getItemData(query: string, headersOrder: string[]): Promise<object> {
-    let url = 'http://next.obudget.org/api/query?query=' +
-      encodeURIComponent(query);
+  private _budgetNumberFormatter(value: any) {
+    return parseFloat(value).toLocaleString('he-IL', { style: 'currency', currency: 'ILS' });
+  }
+
+  private _budgetLinkFormatter(value: string, hLink: string) {
+    return '<a>' + value + '</a>';
+  }
+
+  getItemData(query: string, headersOrder: string[], formatters: string[]): Promise<object> {
+    const url = 'http://next.obudget.org/api/query?query=' + encodeURIComponent(query),
+    formatterKeys = _.keys(formatters),
+    formatterValues = _.values(formatters);
+
     return new Promise<any>((resolve, reject) => {
       this.http.get(url)
         .map((res: any) => res.json())
         .subscribe(
           (res: any) => {
-            let items: object[] = [];
-            let rows = res.rows;
-            let headers = rows.length > 0 ? _.union(headersOrder, _.keys(_.first(rows))) : [];
+            let items: object[] = [],
+            rows = res.rows,
+            headers = rows.length > 0 ? _.union(headersOrder, _.keys(_.first(rows))) : [];
+
             _.each(rows, (row) => {
               let newItem: any[] = [];
+
               _.each(headers, (header) => {
-                newItem.push(row[header]);
+                let item = row[header];
+
+                if (typeof item === 'string' || typeof item === 'number') {
```

Comment:
why the type restriction?

Reply:
for performance issue - when the item is null object just continue to next step

Reply:
so why not check that it's not null i.e. `if (item) { ...`

Reply:
plus, what if some formatter has special treatment for nulls?
for example, number formatter could display '-' for nulls.